pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -82,6 +86,15 @@ type Config struct {
 	// the cloudprovider will try to add all nodes to a single backend pool which is forbidden.
 	// In other words, if you use multiple agent pools (availability sets), you MUST set this field.
 	PrimaryAvailabilitySetName string `json:"primaryAvailabilitySetName" yaml:"primaryAvailabilitySetName"`
+	// Scale mode of azure nodes. Candidate valudes include: vmss and vmas.
+	// If not set, it will be default to vmas.
+	ScaleMode string `json:"scaleMode" yaml:"scaleMode"`
```

Comment:
I feel like "ScaleMode" is a weird name.  "VMType"?

Reply:
LGTM